### ISSUE TITLE
get_budgets returns only the most recent month, not the history for the past year

### ISSUE THREAD

**Issue by ashleydb** [mintapi/mintapi#66]
The json data returned from mint.com contains the full history of budget data, but the output from mintapi only shows the current month.

I'm not familiar with Python, but I suspect it is related to one of these sections of setting up and filling out the return data. I feel like there is a loop missing...

```
    # Make the skeleton return structure
    budgets = {
        'income': response['data']['income'][
            str(max(map(int, response['data']['income'].keys())))
        ]['bu'],
        'spend': response['data']['spending'][
            str(max(map(int, response['data']['income'].keys())))
        ]['bu']
    }

    # Fill in the return structure
    for direction in budgets.keys():
        for budget in budgets[direction]:
            budget['cat'] = self.get_category_from_id(
                budget['cat'],
                categories
            )
```

**Comment by mrfiat** [comment]
Is the MintAPI working for you right now?  I cannot get it to get transactions.  See my bug report.  Thx.

**Comment by ashleydb** [comment]
I didn't try get transactions but the default call, (accounts,) and budgets
worked for me.

On Sunday, 17 January 2016, mrfiat <notifications@github.com> wrote:

> Is the MintAPI working for you right now? I cannot get it to get
> transactions. See my bug report. Thx.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrooney/mintapi/issues/66#issuecomment-172331767>.
>


-- 


Ashley Bennett
www.HiAsh.com

**Comment by spearna** [comment]
I also wanted a budget history so I can trend my budget actuals in my local monthly budget tool.  
I looked into this and worked a solution.  Currently, the 'budgets' dictionary in the api is being created just with the max month number (ie most current). I implemented a 'hist' argument in 'get_budget()' that allows for a select-able range from 1-12 months. 
I have it ready to go, just need to get set up to create a Pull Request. Do I have to be approved somehow to do this?
